firewall/xdp.c

```diff
@@ -186,4 +212,63 @@ int classifier(struct xdp_md *ctx) {
   return XDP_PASS;
 }
 
+static __always_inline bool socket_marked(__u32 address, __u32 port) {
+  struct socket_tuple key = {
+    .address = address,
+    .port = port,
+  };
+  u8 *marked = bpf_map_lookup_elem(&marked_sockets, &key);
+  if (marked) {
```

Comment:
do we need to `bpf_sk_release(marked)` here to avoid leaking a ref count or something?

Reply:
Initially we did since I was using a sockmap  -- a non-nil return value from a map lookup on a sockmap needs to be released, otherwise to avoid leaks the program won't actually load.

We don't need to do the release anymore since I switched things to a regular map since really we only need to check that the socket tuple itself is actually marked (we don't have access to the mark/underlying socket on the `sk_lookup`, so we just check for existence of the tuple in the map we've populated in the sockmap hook).